### PR TITLE
feat: add hot swap for the lang and the searchLang

### PR DESCRIPTION
**Pull request by tiborux**
To test it, you can open the server and open the console. There, you can type:

window.initX.lang = "es"
window.initX.searchLang = "es"

EX-4120

### CODE REVIEW

File: docs/i18n.md [hunk 1]
```diff
@@ -146,17 +147,25 @@ initial bundle size. It will only be downloaded when it is needed.
 
 ### Changing the locale
 
-Because changing a locale may imply downloading new messages, a `setI18nConfiguration` method has been created
-as a helper. This method is exposed after initializing the internationalization with
-`createI18NInstance` in the [src/main.ts](../src/main.ts) file.
+There are some options in the reactive config where you can set the lang and the searchLang.
```
Comment (pmareke) [inline docs/i18n.md:150]: There are some options in the reactive config where you can set the `lang` and the `searchLang`.

File: src/App.vue [hunk 2]
```diff
@@ -63,6 +64,20 @@
     public get currencyFormat(): string {
       return currencies[this.snippetConfig.currency!];
     }
+
+    @Watch('snippetConfig.lang')
+    syncLang(lang: string): void {
+      this.$setLocale(lang);
```
Comment (pmareke) [inline src/App.vue:70]: I think that if you change the `lang` you should change the `searchLang` too, as you are using the lang as searchLang in the `snippet-script.js` file.

File: src/vue.d.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import Vue from 'vue';
```
Comment (pmareke) [inline src/vue.d.ts:1]: Remove this import as it's not used.

File: src/x-components/plugin.options.ts [hunk 4]
```diff
@@ -17,8 +17,7 @@ export const installXOptions: InstallXOptions = {
       messages
     });
     vue.use(i18n);
-    (window as any).setLocale = i18n.setLocale.bind(i18n); // Only for testing
```
Comment (pmareke) [inline src/x-components/plugin.options.ts:20]: I'll leave the `setLocale` in the window as it could be used in some tests.

**Review by pmareke — CHANGES_REQUESTED** [review]

File: docs/i18n.md [hunk 5]
```diff
@@ -146,17 +147,25 @@ initial bundle size. It will only be downloaded when it is needed.
...
+
+- The lang is the value used for the static messages.
```
Comment (CachedaCodes) [inline docs/i18n.md:152]: Remove both `The` and format "lang" and "searchLang" as code.

- [ ] Remove `The` and format as code

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 2 above)
Comment (CachedaCodes) [reply src/App.vue:70]: I understand from the `snippet-script.js` that `searchLang` is dependant on `lang` when `searchLang` doesn't have a set value, so maybe it doesn't apply here. At most I would change both if there's no `searchLang` URL parameter, but still don't see it as necessary. Up to you @tiborux

File: src/vue.d.ts (hunk 3 above)
Comment (tiborux) [reply src/vue.d.ts:1]: if you remove it you will get this error:

image

File: src/x-components/plugin.options.ts (hunk 4 above)
Comment (tiborux) [reply src/x-components/plugin.options.ts:20]: but you can change it using window.initX.lang

Comment (pmareke) [reply src/x-components/plugin.options.ts:20]: If you remove this binds, you caan't right?

**Review by pmareke — APPROVED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Pull request merged**